core/src/main/java/org/bitcoinj/base/internal/ByteUtils.java

```diff
@@ -104,8 +119,43 @@ public static BigInteger bytesToBigInteger(byte[] bytes) {
      *                                        if value doesn't fully fit the buffer
      */
     public static void writeUint16LE(int val, byte[] out, int offset) throws ArrayIndexOutOfBoundsException {
-        out[offset] = (byte) (0xFF & val);
-        out[offset + 1] = (byte) (0xFF & (val >> 8));
+        check(offset >= 0 && offset <= out.length - 2, () ->
+                new ArrayIndexOutOfBoundsException(offset));
+        writeUint16LE(val, ByteBuffer.wrap(out, offset, out.length - offset));
```

Comment:
There's an object creation here,  don't know if there are any use cases where this would affect performance.

Reply:
Tbh I don't see a noticable regression of the unit tests. I'd say we can optimize later, if we run into problems here.